v2/tools/generator/internal/kustomization/crd_kustomization_file.go

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package kustomization
+
+import (
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// CRDKustomizeFile captures the structure of the kustomization.yaml file we want to write
+// All file paths specified as relative paths from the location of kustomize.yaml
+type CRDKustomizeFile struct {
+	Resources []string `yaml:"resources"` // File paths to resource CRDs
+	Patches   []string `yaml:"patches"`   // File paths to patch files used to enable conversions
+}
+
+// NewCRDKustomizeFile creates a new CRDKustomizeFile ready to populate
+func NewCRDKustomizeFile() *CRDKustomizeFile {
+	return &CRDKustomizeFile{}
+}
+
+// AddResource adds a filepath specifying another CRD definition to include
+func (k *CRDKustomizeFile) AddResource(resourceFilePath string) {
+	k.Resources = append(k.Resources, resourceFilePath)
+}
+
+// AddPatch adds a filepath specifying another CRD patch to include
+func (k *CRDKustomizeFile) AddPatch(patchFilePath string) {
+	k.Patches = append(k.Patches, patchFilePath)
+}
```

Comment:
I was thinking maybe merge these into one method since we always want to add a resource and patch at the same time, but thinking about it I guess that would tie this type to this first use case too much.

Reply:
Even though it's a single use type, keeping coupling between methods down seems like a good practice.